sssd_test_framework/roles/ad.py

```diff
@@ -2009,4 +2024,91 @@ def policy(self, logon_rights: dict[str, list[ADObject]], cfg: dict[str, Any] |
         return self
 
 
+class ADPasswordPolicy(BaseObject[ADHost, AD]):
+    """
+    Password policy management.
+    """
+
+    def __init__(self, role: AD):
+        """
+        :param role: AD host object.
+        :type role: ADHost
+        """
+        super().__init__(role)
+
+    def complexity(self, enable: bool) -> ADPasswordPolicy:
+        """
```

Comment:
I'm going to change this back to a `@property` , so it's not instantiated with every call, I forgot about that and I will put usage examples in that docstring.

Reply:
Making this function a property will not allow you to return a `ADPasswordPolicy` object (self) in order to chain  method invocations. I have nothing against it, but it doesn't align with the rest of your code. Do as you want.

Reply:
Ah that explains something I didn't quite understand. I made this decision change because it wouldn't automatically instantiate the object, but I do want the other behavior so I will change this back.